lib/elixir/pages/anti-patterns/design-anti-patterns.md

```diff
@@ -0,0 +1,40 @@
+# Design-related anti-patterns
+
+This document outlines anti-patterns related to your modules, functions, and the role they
+play within a codebase.
+
+## Primitive obsession
+
+TODO
+
+## Boolean obsession
+
+TODO
```

Comment:
New anti-pattern. A specialization about primitive obsession but focused on booleans.

Reply:
> New anti-pattern. A specialization about primitive obsession but focused on booleans.

@josevalim In the description of "Boolean obsession" it would be interesting to make clear an observation you made in a recent email: _"...in this case we are replacing one primitive with another..."_

This observation clearly justifies the main difference between "Primitive obsession" and "Boolean obsession"

Reply:
I like that a lot. :)